### PR TITLE
Some minor handshake improvements

### PR DESCRIPTION
**Pull request by davidkelly**
When some processing is delayed, we could be in a situation where the 2 parties in a handshake are done at somewhat different times.  Now, we detect when the other party has called the backend to create a connection request.  If we were still waiting for the 1.6 seconds till the handshake was 'done', we stop and immediately create our connection request.  If instead we are polling to see if they have created theirs, we reset the polling count, giving us a bit more time to see it connect us.  This _should_ help with some of the handshake issues we saw.  

Also -- added the current hand/joint index to the logging, which will help debugging other issues.

### CODE REVIEW

**Comment by davidkelly** [comment]
Test Plan:

Smoke test the handshake.  You cannot really easily get into the state that exercises this code on purpose, so a smoke is sufficient.  Hit me up and we can do it together -- we will be on different networks so _maybe_ a bit of latency somewhere which will be good to test.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 5b5695663707f63c0c15e503262d3a664b281b6f available at: https://gist.github.com/64e4099c92ae9883e702921b2a7ddcf4<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10384-5b56956.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10384-5b56956.dmg

**Comment by kevinmthomasse** [comment]
testing

File: scripts/system/makeUserConnection.js [hunk 1]
```diff
@@ -768,11 +769,20 @@
...
+                if (state == STATES.MAKING_CONNECTION && connectingId === senderID) {
+                    // we are making connection, they just started, so lets reset the
+                    // poll count just in case
+                    pollCount = 0;
+                } else {
+                    // if waiting or inactive, lets clear the connecting id. If in makingConnection,
+                    // do nothing
                     clearConnecting();
                     if (state !== STATES.INACTIVE) {
                         startHandshake();
```
Comment (howard-stearns) [inline scripts/system/makeUserConnection.js]: In this PR, we do these steps above (clearConnecting.... maybe startHanshake) in one different case that we did not used to, when connectingId !== senderID. 

Is that ok?

**Comment by kevinmthomasse** [comment]
## QA Test
**QA Tester**: KevinMThomas
**QA Pass**: **QA Pass** SUCCESS
**Additional Test Results + Observations**:  Successfully tested with @davidkelly in ```dev-welcome```.  I was utilizing the Oculus with the Touch controllers.  I was successfully able to grip @davidkelly right hand with both my left and right hands.  There is a small issue with the Oculus that is NOT related to this PR as when the controllers are out of range of the camera, unexpected behavior occurs with unpredictable physics occurs where I would assume when out of range the hands would stay in the last sensor view.  Again NOT part of this PR this is a pass.

File: scripts/system/makeUserConnection.js (hunk 1 above)
Comment (davidkelly) [inline scripts/system/makeUserConnection.js]: Oh interesting -- that is a good catch.  That isn't right -- I should not do so in this case.  I'll pop that in.  This of course won't matter when only 2 people test, which is why we saw nothing weird.

Comment (davidkelly) [inline scripts/system/makeUserConnection.js]: Ugh -- this logic is already messy -- I'm going to spend a bit of time looking at it, in hopes of making it more readable.  Nesting the ifs starts becoming hard to read quickly...

**Comment by hifi-gustavo** [comment]
Build console output for commit: a4125c1507e1e7f4defa73969579c70d0ae3ed5d available at: https://gist.github.com/64e4099c92ae9883e702921b2a7ddcf4<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10384-a4125c1.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10384-a4125c1.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: a4125c1507e1e7f4defa73969579c70d0ae3ed5d available at: https://gist.github.com/64e4099c92ae9883e702921b2a7ddcf4<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10384-a4125c1.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10384-a4125c1.dmg

**Pull request merged**